### PR TITLE
add soil-metrics package

### PR DESCRIPTION
**Pull request by jaycenhorton**

### CODE REVIEW

File: packages/ggit/src/__tests__/client.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  describe('Client', () => {
+    describe('constructor', () => {
+      it('will create a soil metrics GGIT API client', () => {
+        expect(new Client()).toBeInstanceOf(Client);
+      });
+    });
+
+    describe('configure', () => {
+      it('will not throw an error when configuring if the credentials are invalid', async () => {
```
Comment (scottynomad) [inline packages/ggit/src/__tests__/client.test.ts:20]: `invalid` should have been `valid` here?

File: packages/ggit/src/__tests__/client.test.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      it('will throw an error when configuring if the credentials are invalid', async () => {
+        mockTokenEndpoint({ throws: true });
+        await expect(new Client().configure(CREDENTIALS)).rejects.toStrictEqual(
+          new Error('{"status":401,"ok":false}')
+        );
+      });
+    });
+
+    describe('createClient', () => {
+      it('will not throw an error when configuring if the credentials are invalid', async () => {
```
Comment (scottynomad) [inline packages/ggit/src/__tests__/client.test.ts:35]: `invalid` -> `valid`

File: packages/ggit/src/fetch.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    body: ApiType['request']['body'];
+    headers: ApiType['request']['headers'];
+  }): Promise<ApiType['response']> {
+    try {
+      const options = this.buildOptions({
+        body,
+        headers,
+      });
+      const response = await fetch(this.uri, options);
+      const data = await response.json();
```
Comment (scottynomad) [inline packages/ggit/src/fetch.ts:56]: This should probably go after the `response.ok` check unless you're passing `Accept: application/json` and their API is better behaved that most and actually returns JSON formatted bodies for all error codes.

**Review by scottynomad — COMMENTED** [review]
Looks great!  A couple of very minor comments ...

**Comment by jaycenhorton** [comment]
> This should probably go after the `response.ok` check unless you're passing `Accept: application/json` and their API is better behaved that most and actually returns JSON formatted bodies for all error codes. 

good call

File: packages/ggit/src/fetch.ts (hunk 3 above)
Comment (jaycenhorton) [reply packages/ggit/src/fetch.ts:56]: good call

**Review by scottynomad — APPROVED** [review]

**Pull request merged**